Build errors after updating to 3.1.6

### Vue - Official extension or vue-tsc version

3.1.6

### VSCode version

Not used

### Vue version

3.5.25

### TypeScript version

5.8.3

### System Info

```shell
System:
    OS: Linux 6.12 EndeavourOS
    CPU: (12) x64 AMD Ryzen 5 5600X 6-Core Processor
    Memory: 11.93 GB / 31.25 GB
    Container: Yes
    Shell: 5.3.3 - /bin/bash
  Binaries:
    Node: 22.17.0 - /home/kubajmarek/.nvm/versions/node/v22.17.0/bin/node
    npm: 10.9.2 - /home/kubajmarek/.nvm/versions/node/v22.17.0/bin/npm
    bun: 1.3.3 - /home/kubajmarek/.bun/bin/bun
  Browsers:
    Chromium: 142.0.7444.175
    Firefox: 145.0.2
    Firefox Developer Edition: 145.0.2
```

### package.json dependencies

```json

```

### Steps to reproduce

Run `vue-tsc --build`

### What is expected?

Clean build without errors, as in v3.1.5

### What is actually happening?

3 errors are produced:

One:
```
TypeError: Cannot read properties of undefined (reading 'length')
    at walkObjectLiteral (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/element.js:307:138)
    at walkObjectLiteral.next (<anonymous>)
    at generateStyleScopedClassReferences (/home/kubajmarek/Documents/code/PROJECTm/node_modules/@vue/language-core/lib/codegen/template/element.js:281:28)
    at generateStyleScopedClassReferences.next (<anonymous>)
    at generateComponent (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/element.js:190:12)
    at generateComponent.next (<anonymous>)
    at generateTemplateChild (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/templateChild.js:48:56)
    at generateTemplateChild.next (<anonymous>)
    at generateElement (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/element.js:237:58)
    at generateElement.next (<anonymous>)
```

Two:
```
TypeError: Cannot read properties of undefined (reading 'length')
    at walkObjectLiteral (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/element.js:307:138)
    at walkObjectLiteral.next (<anonymous>)
    at generateStyleScopedClassReferences (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/element.js:281:28)
    at generateStyleScopedClassReferences.next (<anonymous>)
    at generateElement (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/element.js:233:12)
    at generateElement.next (<anonymous>)
    at generateTemplateChild (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/templateChild.js:44:54)
    at generateTemplateChild.next (<anonymous>)
    at generateVFor (/home/kubajmarek/Documents/code/PROJECT/node_modules/@vue/language-core/lib/codegen/template/vFor.js:57:58)
    at generateVFor.next (<anonymous>)
```

Three:
```
src/components/landing/HeroContent.vue:22:29 - error TS2306: File '/home/kubajmarek/Documents/code/PROJECT/src/components/landing/HeroSection.vue' is not a module.

22 import type { Banner } from "@/components/landing/HeroSection.vue";
```

### Link to minimal reproduction

_No response_

### Any additional comments?

I removed import causing error no. 3 but that didn't influence first 2.

Downgrading package resolves all 3.

--- a/packages/language-core/lib/codegen/template/element.ts
+++ b/packages/language-core/lib/codegen/template/element.ts
@@ -3,7 +3,7 @@ import { camelize, capitalize } from '@vue/shared';
 import { toString } from 'muggle-string';
 import type * as ts from 'typescript';
 import type { Code, VueCodeInformation } from '../../types';
-import { getElementTagOffsets, hyphenateTag, normalizeAttributeValue } from '../../utils/shared';
+import { getElementTagOffsets, getNodeText, hyphenateTag, normalizeAttributeValue } from '../../utils/shared';
 import { codeFeatures } from '../codeFeatures';
 import { createVBindShorthandInlayHintInfo } from '../inlayHints';
 import * as names from '../names';
@@ -382,6 +382,7 @@ function* generateStyleScopedClassReferences(
 			const startOffset = prop.exp.loc.start.offset - 1;
 			const ast = getTypeScriptAST(ts, template, content);
 			const literals: ts.StringLiteralLike[] = [];
+			const identifiers: ts.Identifier[] = [];
 
 			for (const node of forEachNode(ts, ast)) {
 				if (
@@ -396,10 +397,10 @@ function* generateStyleScopedClassReferences(
 					literals.push(expression);
 				}
 				else if (ts.isArrayLiteralExpression(expression)) {
-					yield* walkArrayLiteral(expression);
+					walkArrayLiteral(expression);
 				}
 				else if (ts.isObjectLiteralExpression(expression)) {
-					yield* walkObjectLiteral(expression);
+					walkObjectLiteral(expression);
 				}
 			}
 
@@ -410,29 +411,30 @@ function* generateStyleScopedClassReferences(
 				}
 			}
 
-			function* walkArrayLiteral(node: ts.ArrayLiteralExpression) {
+			for (const identifier of identifiers) {
+				const text = getNodeText(ts, identifier, ast);
+				yield* generateStyleScopedClassReference(template, text, identifier.end - text.length + startOffset);
+			}
+
+			function walkArrayLiteral(node: ts.ArrayLiteralExpression) {
 				const { elements } = node;
 				for (const element of elements) {
 					if (ts.isStringLiteralLike(element)) {
 						literals.push(element);
 					}
 					else if (ts.isObjectLiteralExpression(element)) {
-						yield* walkObjectLiteral(element);
+						walkObjectLiteral(element);
 					}
 				}
 			}
 
-			function* walkObjectLiteral(node: ts.ObjectLiteralExpression) {
+			function walkObjectLiteral(node: ts.ObjectLiteralExpression) {
 				const { properties } = node;
 				for (const property of properties) {
 					if (ts.isPropertyAssignment(property)) {
 						const { name } = property;
 						if (ts.isIdentifier(name)) {
-							yield* generateStyleScopedClassReference(
-								template,
-								name.text,
-								name.end - name.text.length + startOffset,
-							);
+							identifiers.push(name);
 						}
 						else if (ts.isStringLiteral(name)) {
 							literals.push(name);
@@ -445,11 +447,7 @@ function* generateStyleScopedClassReferences(
 						}
 					}
 					else if (ts.isShorthandPropertyAssignment(property)) {
-						yield* generateStyleScopedClassReference(
-							template,
-							property.name.text,
-							property.name.end - property.name.text.length + startOffset,
-						);
+						identifiers.push(property.name);
 					}
 				}
 			}

